### DISCUSSION TITLE
Pre-final release bug testing session by T13-3

### DISCUSSION THREAD

**Discussion by vimalapugazhan** [AY2324S2-CS2113-T13-4/tp#256] [General]
Cooperating with T13-3 to find bugs to further refine app.

**Comment by tayponghee** [comment]
![image](https://github.com/AY2324S2-CS2113-T13-4/tp/assets/110646458/556171ff-a824-409f-9585-c70cebadc1b7)
changed price to negative in the file

**Reply by tayponghee** [reply]
![image](https://github.com/AY2324S2-CS2113-T13-4/tp/assets/110646458/40f9e261-c6bb-4cb1-8029-d73fd43f74c2)
for added bugginess

**Reply by tayponghee** [reply]
![image](https://github.com/AY2324S2-CS2113-T13-4/tp/assets/110646458/c616b9cb-c241-42d3-bbc5-a7ebea3896b0)
quantity can also go negative

**Comment by alalal47** [comment]
![image](https://github.com/AY2324S2-CS2113-T13-4/tp/assets/110593019/9caad1c2-91db-44a5-8dbe-60071cd91bb3)
UG bug, no spacing after line 13 clear:

**Comment by tayponghee** [comment]
![image](https://github.com/AY2324S2-CS2113-T13-4/tp/assets/110646458/10c4fc35-e918-43e7-9185-3391f7fbb18a)
causes the file to throw an exception
while the names can be anything the user wants, adding a bunch of commas in that manner will cause the save file to be corrupted

**Reply by tayponghee** [reply]
![image](https://github.com/AY2324S2-CS2113-T13-4/tp/assets/110646458/8c18cc48-12b5-4de2-a9cb-37f28c9cb76d)
Another example here

**Comment by tayponghee** [comment]
this one doesnt even throw exception when saving it to the list:
![image](https://github.com/AY2324S2-CS2113-T13-4/tp/assets/110646458/7b682be2-3037-42c1-bd40-3518d656c648)

**Comment by tayponghee** [comment]
![image](https://github.com/AY2324S2-CS2113-T13-4/tp/assets/110646458/81195723-930a-4543-a87c-377106dea222)
can change the name to the regex

**Comment by tayponghee** [comment]
![image](https://github.com/AY2324S2-CS2113-T13-4/tp/assets/110646458/6e3815cf-251c-414d-a222-e150f22c4dcc)
same here

**Comment by tayponghee** [comment]
mildly different ways of getting the compiler to accept the new item:
![image](https://github.com/AY2324S2-CS2113-T13-4/tp/assets/110646458/d20596b6-a080-4f24-bda9-2998167887ea)

**Comment by blackmirag3** [comment]
![image](https://github.com/AY2324S2-CS2113-T13-4/tp/assets/78994143/8c5fcbd8-a5c8-4acf-9ee5-104696e2bfbc)
Duplicate items in save file with different stats get overwritten by the latest entry for the same item.
Could consider documenting or show that the programme detects duplicate items in a corrupt save file to convince user what the expected behaviour is

**Comment by blackmirag3** [comment]
Similar to Pong Hee's bug but programme does not throw exception or highlight error
![image](https://github.com/AY2324S2-CS2113-T13-4/tp/assets/78994143/127813a4-ecad-466e-93ae-5dd68b33e3a4)

**Reply by blackmirag3** [reply]
![image](https://github.com/AY2324S2-CS2113-T13-4/tp/assets/78994143/2b4ec768-ca81-4c28-a166-f78cec1673b3)
Name is saved wrongly in data file

Expcted ,,, ,,, ,,, but saved as ,,,*&_,,,
